pkg/apis/serving/v1/revision_defaults.go

```diff
@@ -53,6 +53,11 @@ func (rs *RevisionSpec) SetDefaults(ctx context.Context) {
 		rs.TimeoutSeconds = ptr.Int64(cfg.Defaults.RevisionTimeoutSeconds)
 	}
 
+	// Default IdleTimeoutSeconds based on our configmap.
+	if rs.IdleTimeoutSeconds == nil || *rs.IdleTimeoutSeconds == 0 {
+		rs.IdleTimeoutSeconds = ptr.Int64(cfg.Defaults.RevisionIdleTimeoutSeconds)
+	}
```

Comment:
I think this piece breaks the upgrade test.

Reply:
Yeah setting the default will break downgrade so you'll want to introduce the field without setting/using it and then add it in the following release

there's an issue to make the process simpler https://github.com/knative/serving/issues/11980